### PR TITLE
Add support for TLD endpoints

### LINKED ISSUE #32: Implement TLD endpoints
https://developer.dnsimple.com/v2/tlds/

### PR DESCRIPTION
**Pull request by jacegu**
Closes #32

### CODE REVIEW

**Comment by jacegu** [comment]
That option parsing was trickier than I expected...

**Comment by jacegu** [comment]
Tested this against sandbox and works as expected. Marking as ready for review... 👍

**Comment by jacegu** [comment]
@aeden @weppos if you can take a look at the changes I did to both `Dnsimple.Response` and `Dnsimple.Utils` I'd really appreciate it. Other than that this PR is simply adding more endpoints and a review is not really necessary.

File: lib/dnsimple/utils.ex [hunk 1]
```diff
@@ -1,38 +1,64 @@
 defmodule Dnsimple.Utils do
 
   @doc """
-  Converts the map into a struct.
+  Converts a ap into a struct.
```
Comment (aeden) [inline lib/dnsimple/utils.ex]: Typo

**Comment by aeden** [comment]
@jacegu I have a question about the transformation of maps into structs: given that the result we get back is from poison, and poison supports defining the types for transformation, does it make sense to use poison's transformation logic instead of writing our own?

(Examples of transformation here: https://github.com/devinus/poison/blob/master/test/poison/decoder_test.exs)

**Pull request merged**